src/kademlia/get_item.cpp

```diff
@@ -78,7 +78,7 @@ void get_item::got_data(bdecode_node const& v,
 	// data can reach here, which means pk and sig must be valid.
 	if (!pk || !sig) return;
 
-	std::pair<char const*, int> salt(m_salt.c_str(), int(m_salt.size()));
```

Comment:
Now a temporary copy of the salt is taken for computing the item target id and same is used for assigning too. Thanks for highlighting aliasing.